lib/pbench/test/unit/server/query_apis/conftest.py

```diff
@@ -60,3 +60,61 @@ def query_api(
         return response
 
     return query_api
+
+
+@pytest.fixture
+@responses.activate
+def query_api_get(client, server_config, provide_metadata):
+    """
+    Help controller queries that want to interact with a mocked
+    Elasticsearch service.
+
+    This is a fixture which exposes a function of the same name that can be
+    used to set up and validate a mocked Elasticsearch query with a JSON
+    payload and an expected status.
+
+    Parameters to the mocked Elasticsearch GET are passed as keyword
+    parameters: these can be any of the parameters supported by the
+    responses mock. Exceptions are specified by providing an Exception()
+    instance as the 'body'.
+
+    :return: the response object for further checking
+    """
+
+    def query_api(
+        pbench_uri: str,
+        es_uri: str,
+        payload: Dict[str, Any],
+        expected_index: str,
+        expected_status: str,
+        headers: dict = {},
+        **kwargs,
+    ) -> requests.Response:
+        host = server_config.get("elasticsearch", "host")
+        port = server_config.get("elasticsearch", "port")
+        es_url = f"http://{host}:{port}{expected_index}{es_uri}"
+        with responses.RequestsMock() as rsp:
+            """
+            We need to set up mocks for the Server's call to Elasticsearch,
+            which will only be made if we don't expect Pbench to fail before
+            making the call. We never expect an Elasticsearch call when the
+            expected status is FORBIDDEN or UNAUTHORIZED; or when we give the
+            canonical "bad username" (badwolf) and are expecting NOT_FOUND.
+            """
```

Comment:
I'm finding this text confusing.  For one thing, I think we only set up a single mock, for a request to `es_url`.  (I suppose we set up multiple mocks across multiple calls to this function, but I don't think that perspective helps understanding here.)  It's true that we only set the mock under the circumstances which the text outlines, but whether the CUT calls ES or not has nothing to do with our expectations here.  That is, we don't set up the mock when we expect the CUT to issue a failure before invoking the mock (which is not really what the text says).

Moreover, if we _don't_ expect the Server to call ES, then we should have a mock for it in place which fails the test if it is invoked....